### DISCUSSION TITLE
Confused about the different options to control a heat pump

### DISCUSSION THREAD

**Discussion by lfrancke** [raomin/ESPAltherma#351] [Q&A]
I've been digging into this project and I'm a bit confused.

I do understand that we use a connector labeled X10A to gather information.
This is a serial bus which is also used by the official diagnostics tool.

I also understand that there are various different ways of controlling the heat pumps but I'm not clear on the differences:

- P1/P2
  - Connection used by thermostats?
  - Can monitor and control
  - Seems to require more hardware
- SG1/SG2: Smart Grid, I understand that these are [not available on all heat pumps](https://github.com/raomin/ESPAltherma/issues/3#issuecomment-786938734) and it requires additional relays
  - Described in the [README](https://github.com/raomin/ESPAltherma#step-4-optional---smart-grid-features)
  - More control (4 states)
- "External Thermostat": Apparently an external thermostat can be bought/attached to some/all heatpumps and this can turn the heat pump on or off
  - This seems to be a binary choice: on or off
  - Requires additional relays
  - Described in the [README](https://github.com/raomin/ESPAltherma#step-3-optional---controling-your-daikin-altherma-heat-pump)
- X85A: This is used by an optional addon called "Demand PCB" which can be used by energy management systems to limit the amount of energy the heat pump uses
  -  There are four levels and they can be configured
  - The Daikin board is called EKRP1AHTA
  - Does not require relays? Or does it?

Questions:
- Which of these are recommended to use and give the greatest flexibility?
- Does it make sense to have multiple of these options (e.g. Smart Grid plus "Demand PCB")?

Thank you!

One more question: What kind of manuals do you use to find out the information about which connectors are available where?

**Comment by DieterB61** [comment]
> Which of these are recommended to use and give the greatest flexibility?

What do you need? 
I use the input for an external thermostat with an external thermostat, but I can also close the contact using an additional relay (sonoff switch).
I use X10A to gather information.
And I use X85A to set maximum power: max power reduction helps my heat pump to reduce on/off cycles (the internal control technology seems not to be the best) and I use it for adapting the power to the power coming from my photovoltaics (I use the heat capacity of my old house as energy storage).
I don't need SG because I can control room temperature by my thermostat input and do not store energy in hot water (to heat up my cellar :-) )

> Does it make sense to have multiple of these options (e.g. Smart Grid plus "Demand PCB")?

Not for me.

**Reply by lfrancke** [reply]
Thank you!
I think I'm unsure about the difference between X85A and SG as they seem to serve the same purpose?
And how P1/P2 comes into it I don't know.

I know it's a long shot but would you be willing to help me out and educate me a bit. Happy to compensate as well as you did exactly what I'd like to do as well. If you are: my email is in my profile here on GitHub.
